pyproject.toml

```
@@ -142,15 +142,44 @@ Discussions = "https://github.com/ansys/pyaedt/discussions"
 Releases = "https://github.com/ansys/pyaedt/releases"
 Changelog = "https://github.com/ansys/pyaedt/blob/main/doc/source/changelog.rst"
 
-[tool.black]
+[tool.ruff]
 line-length = 120
+exclude = [
+    "venv", 
+    "__init__.py", 
+    "doc/_build", 
+    "doc/source/examples"
+]
+
+[tool.ruff.format]
+quote-style = "double"
+indent-style = "space"
+
+[tool.ruff.lint]
+select = [
+    "I", # isort, see https://docs.astral.sh/ruff/rules/#isort-i
+    "W", # pycodestyle, see https://docs.astral.sh/ruff/rules/#pycodestyle-e-w
+]
+ignore = [
+    "D",   # pydocstyle, see https://docs.astral.sh/ruff/rules/#pydocstyle-d
+    "E",   # pycodestyle, see https://docs.astral.sh/ruff/rules/#pycodestyle-e-w
+    "F",   # pyflakes, see https://docs.astral.sh/ruff/rules/#pyflakes-f
+    "N",   # pep8-naming, see https://docs.astral.sh/ruff/rules/#pep8-naming-n
+    "PTH", # flake8-use-pathlib, https://docs.astral.sh/ruff/rules/#flake8-use-pathlib-pth
+    "TD",  # flake8-todos, https://docs.astral.sh/ruff/rules/#flake8-todos-td
+]
```

```suggestion
select = [
    "I", # isort, see https://docs.astral.sh/ruff/rules/#isort-i
    "W", # pycodestyle, see https://docs.astral.sh/ruff/rules/#pycodestyle-e-w
    "D",   # pydocstyle, see https://docs.astral.sh/ruff/rules/#pydocstyle-d
    "E",   # pycodestyle, see https://docs.astral.sh/ruff/rules/#pycodestyle-e-w
    "F",   # pyflakes, see https://docs.astral.sh/ruff/rules/#pyflakes-f
    "N",   # pep8-naming, see https://docs.astral.sh/ruff/rules/#pep8-naming-n
    "PTH", # flake8-use-pathlib, https://docs.astral.sh/ruff/rules/#flake8-use-pathlib-pth
    "TD",  # flake8-todos, https://docs.astral.sh/ruff/rules/#flake8-todos-td
]
ignore = [fill this list]
```
Everything should be tracked and you should specify all the rules that we need to ignore. This will provide an exhaustive list that we can deal with one item at a time.

---

Exhaustive list of excluded linting rules is provided in the ``pyproject.toml`` file updated in commit XXX